common/diagnostics/Metrics.java

```diff
@@ -113,32 +114,66 @@ private static String formatDateTime(LocalDateTime dateTime) {
         return dateTime.truncatedTo(ChronoUnit.SECONDS).toString();
     }
 
-    protected JsonObject asJSON(boolean reporting) {
-        boolean addUsageDiagnostics = reporting && !base.getReportingEnabled();
+    protected JsonObject asReportingJSON() {
+        return base.getReportingEnabled()
+                ? asFullReportingJSON()
+                : asMinimalReportingJSON();
+    }
 
-        JsonObject metrics = base.asJSON();
-        metrics.add("server", serverProperties.asJSON(!addUsageDiagnostics));
+    protected JsonObject asFullReportingJSON() {
+        JsonObject metrics = base.asReportingJSON();
+        metrics.add("server", serverProperties.asReportingFullJSON());
```

Comment:
This should be `serverProperties.asFullReportingJSON()`. Ditto for the `Minimal` version below.

Reply:
Yep, I've missed the renaming